strongbox-web-core/src/main/java/org/carlspring/strongbox/app/boot/BootProgressBeanPostProcessor.java

```diff
@@ -0,0 +1,125 @@
+package org.carlspring.strongbox.app.boot;
+
+import javax.annotation.PreDestroy;
+import javax.servlet.ServletRegistration;
+
+import io.reactivex.Observable;
+import io.reactivex.subjects.BehaviorSubject;
+import io.reactivex.subjects.Subject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.BeanNameAware;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+import org.springframework.beans.factory.support.DefaultListableBeanFactory;
+import org.springframework.boot.web.embedded.jetty.JettyServletWebServerFactory;
+import org.springframework.boot.web.server.WebServer;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.context.event.ContextRefreshedEvent;
+import org.springframework.context.event.EventListener;
+import org.springframework.core.annotation.Order;
+import org.springframework.lang.Nullable;
+import org.springframework.stereotype.Component;
+import static org.springframework.core.Ordered.HIGHEST_PRECEDENCE;
+
+@Component
+@Order(HIGHEST_PRECEDENCE)
+public class BootProgressBeanPostProcessor
+        implements BeanPostProcessor, BeanNameAware, ApplicationContextAware
+{
+
+    private static final Logger logger = LoggerFactory.getLogger(BootProgressBeanPostProcessor.class);
+
+    private static final Subject<String> beans = BehaviorSubject.create();
+
+    private ApplicationContext applicationContext;
+
+    private String beanName;
```

Comment:
this is a state object accessible by many Threads, so in my opinion it should be marked as `volatile`